src/Features/CSharp/Portable/CodeRefactorings/AddNamedArguments/CSharpAddNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,221 @@
+                    var conditional = (ConditionalAccessExpressionSyntax)node;
+                    return conditional.WhenNotNull is ElementBindingExpressionSyntax;
+
+                default:
+                    return false;
+            }
+        }
+
+        protected override bool TryGetArguments(SyntaxNode node, SemanticModel semanticModel, out SeparatedSyntaxList<ArgumentSyntax> arguments)
+        {
+            switch (node.Kind())
+            {
+                case SyntaxKind.InvocationExpression:
+                    {
+                        var invocation = (InvocationExpressionSyntax)node;
+                        arguments = invocation.ArgumentList.Arguments;
+                        return true;
+                    }
+
+                case SyntaxKind.ElementAccessExpression:
+                    {
+                        var elementAccess = (ElementAccessExpressionSyntax)node;
+                        if (!IsArray(semanticModel, elementAccess.Expression))
+                        {
+                            arguments = elementAccess.ArgumentList.Arguments;
+                            return true;
+                        }
+
+                        arguments = default(SeparatedSyntaxList<ArgumentSyntax>);
+                        return false;
+                    }
+
+                case SyntaxKind.ObjectCreationExpression:
+                    {
+                        var objectCreation = (ObjectCreationExpressionSyntax)node;
+                        arguments = objectCreation.ArgumentList.Arguments;
+                        return true;
+                    }
+
+                case SyntaxKind.ConditionalAccessExpression:
+                    {
+                        var conditional = (ConditionalAccessExpressionSyntax)node;
+                        if (!IsArray(semanticModel, conditional.Expression))
+                        {
+                            var elementBinding = (ElementBindingExpressionSyntax)conditional.WhenNotNull;
+                            arguments = elementBinding.ArgumentList.Arguments;
+                            return true;
+                        }
+
+                        arguments = default(SeparatedSyntaxList<ArgumentSyntax>);
+                        return false;
+                    }
+
+                case SyntaxKind.BaseConstructorInitializer:
+                case SyntaxKind.ThisConstructorInitializer:
+                    {
+                        var constructorInitializer = (ConstructorInitializerSyntax)node;
+                        arguments = constructorInitializer.ArgumentList.Arguments;
+                        return true;
+                    }
+
+                default:
+                    arguments = default(SeparatedSyntaxList<ArgumentSyntax>);
+                    return false;
+            }
+        }
+
+        protected override SyntaxNode ReplaceArgumentList(SyntaxNode node, SeparatedSyntaxList<ArgumentSyntax> argumentList)
```

Comment:
I would actually like this to be on SyntaxGenerator.  But you don't need to do that.